src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -76,11 +76,13 @@ public override ITestEngineRunner MakeTestRunner(TestPackage package)
                     projectCount++;
             }
 
+            ProcessModel processModel = GetTargetProcessModel(package);
+
             // If we have multiple projects or a project plus assemblies
             // then defer to the AggregatingTestRunner, which will make
             // the decision on a file by file basis so that each project
             // runs with its own settings.
-            if (projectCount > 1 || projectCount > 0 && assemblyCount > 0)
+            if (projectCount > 1 || (projectCount > 0 && assemblyCount > 0) || (projectCount > 0 && processModel == ProcessModel.Default))
```

Comment:
@jnm2 Could you clarify what you mean by integration tests? We have had a long-standing discussion about creating integration tests separate from the unit tests we already have, possibly in a separate repository. I don't think that should be done as part of a bug fix like this, but as a separate thing, because it's a pretty large project. OTOH, some tests that are slightly more integrated than what we have already, run as a part of the unit tests, would fit in well here.